nuxeo-core/nuxeo-core-api/src/test/java/org/nuxeo/ecm/core/api/security/TestACP.java

```diff
@@ -388,10 +390,23 @@ public void testMultipleNewPermissionsWithBlockInheritance() {
 
     }
 
+    @Test
+    @Deploy("org.nuxeo.ecm.core.api.tests:OSGI-INF/test-legacy-acp-behavior.xml")
+    public void testLegacyACLBehavior() {
```

Comment:
Could you add a set of tests structured like this:

    @Deploy(legacy-behavior)
    testLegacyACLBehavior() {
        doTestACLBehavior(true);
    }

    @Deploy(current-behavior)
    testCurrentACLBehavior() {
        doTestACLBehavior(false);
    }

    doTestACLBehavior(boolean legacy) {
        ... set up ACP/ACL/ACE etc. ...
        if (legacy) {
            ... expect something ...
        } else {
            ... expect something else ...
        }
    }

Reply:
Done.